include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -559,23 +574,70 @@ bool
 Parameters::viscous_problem() const
 {
   return (equation_type == EquationType::Stokes || equation_type == EquationType::NavierStokes ||
-          use_turbulence_model == true);
+          turbulence_model_data.is_active);
+}
+
+bool
+Parameters::viscous_term_is_linear() const
+{
+  return not viscous_term_is_nonlinear();
+}
+
+bool
+Parameters::viscous_term_is_nonlinear() const
+{
+  return (viscosity_is_variable() &&
+          treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Implicit);
+}
+
+bool
+Parameters::viscosity_is_variable() const
+{
+  return turbulence_model_data.is_active || generalized_newtonian_model_data.is_active;
+}
+
+bool
+Parameters::convective_term_is_treated_implicitly() const
```

Comment:
This function name is also somewhat ambigous: Compare this function to the function `viscous_term_is_nonlinear()` above. As a user, I cannot know why the present function includes `if(convective_problem())`, while the function above does not include something like `if(viscous_problem())`. The problem is that both function names include the  naming `term` but are implemented differently.

Hence, I suggest a renaming of this function to `implicit_convective_problem()`. The name `problem` implies that we check whether the operator is active at all. On top of this, we check whether the treatment is implicit. This is consistent in the sense that functions with a naming including `term` (like `viscous_term_is_...()`) do not contain if-statements whether the operator is active, they just check for a given term of a PDE whether it is nonlinear/implicit depending on the parameters of this term.